### PR TITLE
init the service with its tests

### PR DESCRIPTION
**Pull request by mgamal92**
| Q             | A                                                                  |
|---------------|--------------------------------------------------------------------|
| Branch?       | main                                              |
| Bug fix?      | yes                                                             |
| New feature?  | yes                  |
| Deprecations? | no  |
| Tickets       | Fix #...            |
| License       | MIT                                                                |

### CODE REVIEW

**Comment by mgamal92** [comment]
@imdhemy 
My opinion is to use the service like ```Extractor::getHashtags```, not needed to initialize a new object for now. What do you think ?

**Comment by imdhemy** [comment]
We can allow both dynamic and static calls. Let's stick to:

```php
$extractor = new Extractor();
$hashtags = $extractor->extract($text);
```

And we can also allow static calls through `__callStatic`. This way we can have a testable class and a static facade. I prefer `extract()` over `getHastags()` because it's shorter and we already know we're dealing with hashtags.

File: tests/ExampleTest.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (imdhemy) [inline tests/ExampleTest.php:1]: It would be better to remove this example.

File: tests/ExtractorTest.php [hunk 2]
```diff
@@ -2,103 +2,76 @@
 
 declare(strict_types=1);
 
-namespace Piscibus\PhpHashtag\Tests;
-
 use Piscibus\PhpHashtag\Extractor;
+use function PHPUnit\Framework\assertEquals;
```
Comment (imdhemy) [inline tests/ExtractorTest.php:6]: To follow Piscibus convention, let's rename `$extractor` into `$sut`. This makes it clear that this is the **s**ervice **u**nder **t**est.

File: tests/Pest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
```
Comment (imdhemy) [inline tests/Pest.php:1]: Do we need to keep this file?

**Review by imdhemy — CHANGES_REQUESTED** [review]

File: src/Extractor.php [hunk 4]
```diff
@@ -6,4 +6,18 @@
 
 class Extractor
```
Comment (imdhemy) [inline src/Extractor.php:7]: Could you please add the `__callStatic()` implementation to allow static calls?

**Comment by imdhemy** [comment]
@mgamal92 Could you please check the unresolved comments?

File: src/Extractor.php (hunk 4 above)
Comment (mgamal92) [reply src/Extractor.php:7]: > Could you please add the `__callStatic()` implementation to allow static calls?

It's done @imdhemy

File: tests/ExampleTest.php (hunk 1 above)
Comment (imdhemy) [reply tests/ExampleTest.php:1]: @mgamal92 I meant this

File: tests/ExtractorTest.php (hunk 2 above)
Comment (imdhemy) [reply tests/ExtractorTest.php:5]: @mgamal92 and this one

**Comment by imdhemy** [comment]
@mgamal92 

- https://github.com/piscibus/php-hashtag/pull/1#discussion_r1059891497
- https://github.com/piscibus/php-hashtag/pull/1#discussion_r1059892636

**Comment by mgamal92** [comment]
OK. I will check

File: src/Extractor.php [hunk 5]
```diff
@@ -6,4 +6,26 @@
...
 {
+    /**
+     * Extract the hashtag from a string.
+     */
+    public static function extract(string $text): array
```
Comment (imdhemy) [inline src/Extractor.php:12]: As per our [discussion](https://github.com/piscibus/php-hashtag/pull/1#issuecomment-1368409282) we need to support both dynamic and static calls. Remove the `static` keyword, please.

File: src/Extractor.php [hunk 6]
```diff
@@ -6,4 +6,26 @@
...
+            $hashtags[] = $match;
+        }
+
+        return $hashtags;
+    }
+
+    /**
+     * Enable static calls.
+     */
+    public static function __callStatic(string $name, array $arguments)
```
Comment (imdhemy) [inline src/Extractor.php:27]: After removing the `static` keyword from the `extract()` method, instantiating a new object will be required.

**Review by imdhemy — CHANGES_REQUESTED** [review]

File: src/Extractor.php (hunk 5 above)
Comment (mgamal92) [reply src/Extractor.php:12]: @imdhemy 
Please take a look at [this](https://github.com/phpstan/phpstan/issues/5782)

File: src/Extractor.php (hunk 6 above)
Comment (mgamal92) [reply src/Extractor.php:27]: https://github.com/piscibus/php-hashtag/pull/1#discussion_r1061003019

File: src/Extractor.php (hunk 5 above)
Comment (imdhemy) [reply src/Extractor.php:12]: I didn't get the point

Comment (mgamal92) [reply src/Extractor.php:12]: @imdhemy if the ```static``` keyword removed from the method, the method won't be called statically.
Also I'm sorry for the above link, it seems confusing and no to the point. Instead take a look to this [answer](https://stackoverflow.com/a/70347602/5810944)

Comment (imdhemy) [reply src/Extractor.php:12]: Noted

**Review by imdhemy — APPROVED** [review]
Thank you 🙌

**Pull request merged**